Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -75,19 +78,34 @@ def livecheck_formulae(formulae_to_check, args)
           EOS
         end
 
-        skip_result = skip_conditions(formula, args: args)
+        skip_result = skip_conditions(formula_or_cask, args: args)
         next skip_result if skip_result != false
 
-        formula.head.downloader.shutup! if formula.head?
+        formula.head.downloader.shutup! if formula&.head?
 
-        current = if formula.head?
-          formula.any_installed_version.version.commit
+        livecheck_version = formula_or_cask.livecheck.version
+        current = if livecheck_version.is_a?(String)
```

Comment:
I can't leave a suggestion for this entire block since the `else` is unchanged, unfortunately.

I feel like this is more complicated than it needs to be and we can also avoid a level of nesting if we did something like this:

```ruby
current = if livecheck_version.is_a?(String)
  Version.new(livecheck_version)
elsif formula&.head?
  formula.any_installed_version.version.commit
elsif livecheck_version.is_a?(Array)
  separator, method = livecheck_version
  Version.new(formula_or_cask.version.to_s.split(separator, 2).try(method))
else
  Version.new(formula_or_cask.version)
end
```

We don't want a non-string `livecheck.version` to manipulate a `head` version (i.e., a commit hash), so I've placed it before the `livecheck_version.is_a?(Array)` condition.

I've been looking at this PR for a good while now and my brain's tired, so do check my work to make sure it's sane.

Reply:
> I've been looking at this PR for a good while now and my brain's tired

Your attention to detail is quite remarkable, I must say